### PR TITLE
Support for report parameter dependencies

### PR DESCRIPTION
**Pull request by pnaw94**
UI part for recent requisition service update: https://github.com/OpenLMIS/openlmis-requisition/pull/21

### CODE REVIEW

File: src/report/report-generate.controller.js [hunk 1]
```diff
@@ -113,5 +130,44 @@
...
+
+
+        /**
+         * @ngdoc method
+         * @methodOf report.controller:ReportGenerateController
+         * @name watchDependency
+         *
+         * @description
+         * Sets up a watch on report parameter selection,
+         * to update dependent parameters options based on it's value.
```
Comment (matkwiatkowski) [inline src/report/report-generate.controller.js:138]: Please add @param annotation

File: src/report/report-generate.controller.js [hunk 2]
```diff
@@ -59,6 +63,17 @@
         /**
          * @ngdoc property
          * @propertyOf report.controller:ReportGenerateController
+         * @name reportFactory
+         * @type {Object}
+         *
+         * @description
+         * The object representing the reports factory.
+         */
+        vm.reportFactory = reportFactory;
```
Comment (ngraczewski) [inline src/report/report-generate.controller.js]: Why are we exposing this to the view?

File: src/report/report.factory.js [hunk 3]
```diff
@@ -194,8 +195,11 @@
                                           provided, the value will be used as a display name.
          * @return {Promise}              The promise for report params.
          */
```
Comment (matkwiatkowski) [inline src/report/report.factory.js]: you should remove 1 parameter from doc

File: src/report/report-generate.controller.js [hunk 4]
```diff
@@ -30,14 +30,18 @@
...
+        'reportParamsOptions', 'reportUrlFactory', 'accessTokenFactory'
     ];
 
-    function controller($state, $window, report, reportParamsOptions, reportUrlFactory,
-                        accessTokenFactory) {
+    function controller($state, $scope, $window, report, reportFactory,
+                        reportParamsOptions, reportUrlFactory, accessTokenFactory) {
         var vm = this;
 
+        $scope.vm = vm;
```
Comment (ngraczewski) [inline src/report/report-generate.controller.js]: This is redundant.

File: src/report/report-generate.controller.js [hunk 5]
```diff
@@ -80,6 +95,8 @@
          */
         vm.selectedParamsOptions = {};
 
+        vm.selectedParamsDependencies = {};
```
Comment (ngraczewski) [inline src/report/report-generate.controller.js]: Lacks JSDocs.

File: src/report/report-generate.controller.js [hunk 6]
```diff
@@ -113,5 +130,44 @@
...
+        /**
+         * @ngdoc method
+         * @methodOf report.controller:ReportGenerateController
+         * @name watchDependency
+         *
+         * @description
+         * Sets up a watch on report parameter selection,
+         * to update dependent parameters options based on it's value.
+         */
+        function watchDependency(param, dep) {
```
Comment (ngraczewski) [inline src/report/report-generate.controller.js]: Could this functionality be achieved by ng-change funtion?

Comment (pnaw94) [reply src/report/report-generate.controller.js]: Well, when looping through each parameter, we don't know whether the evaluated parameter is a dependency for any other parameter. What we do know is what are dependencies for a given parameter, and I believe assigning those in html code would be at best very messy.

**Comment by pnaw94** [comment]
@ngraczewski @mkwiatkowskisoldevelo updated

**Pull request merged**